src/backend/base/langflow/components/vectorstores/MongoDBAtlasVector.py

```diff
@@ -106,14 +83,31 @@ def _build_mongodb_atlas(self) -> MongoDBAtlasVectorSearch:
         return vector_store
 
     def search_documents(self) -> List[Data]:
+        from typing import Optional, Union
+        from bson import ObjectId
+        from langchain.schema import Document
+        import json
+
         vector_store = self._build_mongodb_atlas()
 
-        if self.search_input and isinstance(self.search_input, str) and self.search_input.strip():
+        class JSONEncoder(json.JSONEncoder):
+            def default(self, o):
+                if isinstance(o, Union[ObjectId, Document]):
+                    return str(o)
```

Comment:
Document is JSON serializable and ObjectId might not be but there's this: https://stackoverflow.com/a/18405626